modules/navier_stokes/src/functormaterials/LinearFrictionFactorFunctorMaterial.C

```diff
@@ -0,0 +1,84 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LinearFrictionFactorFunctorMaterial.h"
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", LinearFrictionFactorFunctorMaterial);
+
+InputParameters
+LinearFrictionFactorFunctorMaterial::validParams()
+{
+  InputParameters params = FunctorMaterial::validParams();
+  params.addClassDescription("Material class used to compute a friction factor of the form A * "
+                             "f(t) + B * g(t) * |v_I| with A, B vector constants, f(t) and g(t) "
+                             "functions of time, and |v_I| the interstitial speed");
+
+  params.addRequiredParam<MooseFunctorName>("functor_name",
+                                            "The name of functor storing the friction factor");
+  params.addRequiredParam<MooseFunctorName>(NS::porosity, "porosity");
+  params.addParam<MooseFunctorName>(
+      NS::superficial_velocity_x, 0, "The x component of the fluid superficial velocity variable.");
+  params.addParam<MooseFunctorName>(
+      NS::superficial_velocity_y, 0, "The y component of the fluid superficial velocity variable.");
+  params.addParam<MooseFunctorName>(
+      NS::superficial_velocity_z, 0, "The z component of the fluid superficial velocity variable.");
+  params.addParam<RealVectorValue>(
+      "A", RealVectorValue(1, 1, 1), "Coefficient of the A * f(t) term");
+  params.addParam<RealVectorValue>(
+      "B", RealVectorValue(1, 1, 1), "Coefficient of the B * g(t) * |v_I| term");
```

Comment:
no defaults. or defaults of 0

too risky if a user wants linear and we re adding them a quadratic by default